### PR TITLE
[WIP] Modifying SAC to make \alpha converge faster

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

In gray what is on main and in blue with the proposed changes. The entropy converges slightly faster as expected. This leads to a higher cumulative reward at the beginning of training but it seems the final reward is not as high (probably the entropy is not high enough to leave a local minima)

Screen Shot 2021-05-24 at 10 34 30
Screen Shot 2021-05-24 at 10 33 55
Screen Shot 2021-05-24 at 10 33 49

We could set a slower learning rate for the entropy coefficient update to mitigate this issue (it is currently the same as the trainer's like in the original SAC paper)

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/sac/optimizer_torch.py [hunk 1]
```diff
@@ -199,13 +199,9 @@ def __init__(self, policy: TorchPolicy, trainer_params: TrainerSettings):
...
-        self.policy_optimizer = torch.optim.Adam(
-            policy_params, lr=hyperparameters.learning_rate
-        )
-        self.value_optimizer = torch.optim.Adam(
-            value_params, lr=hyperparameters.learning_rate
-        )
-        self.entropy_optimizer = torch.optim.Adam(
+        self.policy_optimizer = torch.optim.Adam(policy_params)
+        self.value_optimizer = torch.optim.Adam(value_params)
+        self.entropy_optimizer = torch.optim.SGD(
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/sac/optimizer_torch.py:204]: SGD makes the entropy_coef converge a lot faster than when using Adam. Since there is only one parameter to optimize, SGD is enough.

File: ml-agents/mlagents/trainers/sac/optimizer_torch.py [hunk 2]
```diff
@@ -601,7 +597,6 @@ def update(self, batch: AgentBuffer, num_sequences: int) -> Dict[str, float]:
         total_value_loss.backward()
         self.value_optimizer.step()
 
-        ModelUtils.update_learning_rate(self.entropy_optimizer, decay_lr)
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/sac/optimizer_torch.py:604]: We should not decay the learning rate for the entropy_coeficient optimizer.

File: ml-agents/mlagents/trainers/sac/optimizer_torch.py [hunk 3]
```diff
@@ -199,13 +199,9 @@ def __init__(self, policy: TorchPolicy, trainer_params: TrainerSettings):
             1e-10,
             self.trainer_settings.max_steps,
         )
-        self.policy_optimizer = torch.optim.Adam(
-            policy_params, lr=hyperparameters.learning_rate
-        )
-        self.value_optimizer = torch.optim.Adam(
-            value_params, lr=hyperparameters.learning_rate
-        )
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/sac/optimizer_torch.py:207]: adding the learning rate as argument to Adam is misleading because these two optimizers use a learning rate schedule that overrides this value anyway.
